src/routeGeneration/templateHelpers.ts

```diff
@@ -128,18 +128,17 @@ export class ValidationService {
       }
     }
 
-    Object.keys(value).forEach(key => {
-      if (!nestedProperties[key]) {
-        if (additionalProperties && additionalProperties !== true) {
-          return this.ValidateParam(additionalProperties, value[key], key, fieldErrors, parent + name + '.', swaggerConfig);
-        } else {
-          return key;
-        }
+    Object.keys(nestedProperties).forEach(key => {
+      const validatedProp = this.ValidateParam(nestedProperties[key], value[key], key, fieldErrors, parent + name + '.', swaggerConfig);
+      if (validatedProp) {
+        value[key] = validatedProp;
       }
-
-      return this.ValidateParam(nestedProperties[key], value[key], key, fieldErrors, parent + name + '.', swaggerConfig);
     });
 
+    if (Object.keys(fieldErrors).length > 0) {
+      return;
```

Comment:
@WoH I approved this PR because you wanted to get it out for people to review, but if you don't mind, please submit a new PR with a comment between lines 138 and 139 that explains why we're not doing anything with the errors. I'll take any code comment at all. When you're scanning the code it's just rather strange.